Flint/src/main/java/flint/CLIAdapter.java

```diff
@@ -1,65 +1,193 @@
+    } catch (Exception e) {
+      // not sure what kind of exceptions will happen
+    } finally {
+      System.exit(0);
     }
 
-    /**
-     * This function checks if filePath leads to an existing java file.
-     *
-     * @param filePath - The file path to check.
-     * @return Returns the true if the file exists and is a java file. Otherwise false.
-     * */
-    private static boolean isJava(String filePath) throws Exception {
-        throw new Exception("Not Yet Implemented");
+    resultOutput(result);
+  }
+
+  /**
+  * This function initializes and returns the FlintConfiguration class.
+  *
+  * @param configPath - The path that leads to the FlintConfiguration subclass jar.
+  * @param className - The java style class name for the FlintConfiguration subclass.
+  * @return Returns the initialized FlintConfiguration subclass if passed in a valid path.
+  * @throws FileNotFoundException - If configPath did not lead to a jar.
+  * @throws IllegalArgumentException - If the passed in configPath is not a FlintConfiguration subclass.
+  * @throws ClassNotFoundException - If the className could not be found
+  * */
+  public static FlintConfiguration configInit(String configPath, String className) throws Exception {
+    File file = new File(configPath);
+
+    if (!file.exists()) {
+      throw new FileNotFoundException();
+    }
+
+    String realConfigPath = configPath;
+
+    if (realConfigPath.charAt(0) != '\\' && realConfigPath.charAt(0) != '/') {
+      // if not passing absolute path
+      realConfigPath = "file://" + System.getProperty("user.dir") + "/" + realConfigPath;
+    } else {
+      realConfigPath = "file://" + realConfigPath;
+    }
+
+    URLClassLoader urlClassLoader = URLClassLoader.newInstance(new URL[] {
+            new URL(
+                    realConfigPath
+            )
+    });
+
+    Class outputClass = urlClassLoader.loadClass(className);
+
+    if (!outputClass.getSuperclass().getSimpleName().equals("FlintConfiguration")) {
+      throw new IllegalArgumentException();
     }
 
-    /**
-     * Starts the Driver to run Flint on fileToLint with config.
-     *
-     * @param fileToLint - The file path that leads to a valid and existing Java file.
-     * @param config - The configuration to run on flieToLint.
-     * */
-    private static Collection<LintFailure> run(String fileToLint, FlintConfiguration config) throws Exception{
-        throw new Exception("Not Yet Implemented");
+    FlintConfiguration output = (FlintConfiguration)outputClass.newInstance();
+
+    return output;
+  }
+
+  /**
+  * This function checks if filePath leads to an existing java file.
+  *
+  * @param filePath - The file path to check.
+  * @return Returns the true if the file exists and is a java file. Otherwise false.
+  * */
+  public static boolean isJava(String filePath) {
+    File file = new File(filePath);
+
+    return Pattern.matches(".*\\.java$", filePath) && file.exists();

```

Comment:
Again, I'm concerned with using `file.exists()` on a filepath that is possibly going to be relative to the directory their command line is in...

Reply:
I thought this is how command lines work? The relative path you entered is always relative to the pwd

Reply:
All right, I had some funny behavior going on when I was testing the driver so I speculated that it wasn't behaving as I thought, but perhaps it was caused by something else.

Reply:
I have open an issue #11 to keep track of the behavior in the future